Mathlib/Data/ZMod/Basic.lean

```diff
@@ -153,19 +153,19 @@ See `ZMod.castHom` for a bundled version. -/
   | _ + 1 => fun i => i.val
 #align zmod.cast ZMod.cast
 
--- see Note [coercion into rings]
-instance (priority := 900) (n : ℕ) : CoeTC (ZMod n) R :=
-  ⟨cast⟩
+-- -- see Note [coercion into rings]
+-- instance (priority := 900) (n : ℕ) : CoeTC (ZMod n) R :=
+--   ⟨cast⟩
 
 @[simp]
-theorem cast_zero : ((0 : ZMod n) : R) = 0 := by
+theorem cast_zero : (cast (0 : ZMod n) : R) = 0 := by
```

Comment:
```suggestion
theorem cast_zero : (cast 0 : R) = 0 := by
```

Reply:
I think you should be able to remove some `ZMod n` annotations now, also more below.